cns/restserver/internalapi_test.go

```diff
@@ -359,6 +397,29 @@ func validateNCStateAfterReconcile(t *testing.T, ncRequest *cns.CreateNetworkCon
 	}
 }
 
+func createNCReqInternal(t *testing.T, secondaryIPConfigs map[string]cns.SecondaryIPConfig, ncID, ncVersion string) cns.CreateNetworkContainerRequest {
+	req := generateNetworkContainerRequest(secondaryIPConfigs, ncID, ncVersion)
+	returnCode := svc.CreateOrUpdateNetworkContainerInternal(req, fakes.NewFakeScalar(releasePercent, requestPercent, batchSize), fakes.NewFakeNodeNetworkConfigSpec(initPoolSize))
+	if returnCode != 0 {
+		t.Fatalf("Failed to createNetworkContainerRequest, req: %+v, err: %d", req, returnCode)
+	}
+	return req
+}
+
+func validateSecondaryIPsNCVersion(t *testing.T, req cns.CreateNetworkContainerRequest) {
+	containerStatus := svc.state.ContainerStatus[req.NetworkContainerid]
```

Comment:
This function seems generic but inside it has hardcoded the ipaddress and expected version. Can we pass the expected SecondaryIpConfig as parameter and compare the req with expected list? This will allow to reuse this function in future as well

Reply:
We can but I don't think it's necessary. I don't think any other test should reuse this function. This validation is designed for TestCreateAndUpdateNCWithSecondaryIPNCVersion specifically.

Reply:
I moved this validation to test function and avoid confusion.